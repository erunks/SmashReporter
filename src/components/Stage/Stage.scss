.stage {
  display: flex;
  position: relative;
}

.stageCard {
  position: relative;
  max-width: 325px;

  .namePlate {
    background-color: black;
    height: 30px;
    max-width: 100%;
  }

  img {
    border: 3px solid;
    height: auto;
    max-height:100%;
    max-width:100%;
    width: 150px;
  }

  .name {
    color: white;
    font-size: 12px;
    padding: 0.25em 2px;
    margin: 0;
    overflow: hidden;
    text-align: center;
    text-overflow: ellipsis;
    white-space: nowrap;
    width: 100%;
  }
}