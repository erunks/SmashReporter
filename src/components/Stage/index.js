import React, { useRef, useState } from 'react';
import propTypes from 'prop-types';
import { useRecoilState } from 'recoil';
import { selectedStageState } from 'recoils/stage';
import { buttonize } from 'helpers/jsx-a11y-helper'
import { IKImage } from 'imagekitio-react';
import { URLS } from 'lib/imageAssets';

import './Stage.scss';

const Stage = (props) => {
  const {
    dlc,
    displayName,
    id,
    name
  } = props;

  const [selectedStage, setSelectedStage] = useRecoilState(selectedStageState);
  const [clickPostion, setClickPosition] = useState({ x: 50, y: 27 });
  const ref = useRef(null);

  const renderStageName = () => (
    <div className="namePlate">
      <div className="name">{ name }</div>
    </div>
  );

  const isStageSelected = () => id === selectedStage;
  const onClick = (event) => {
    setSelectedStage(id);
    const { left, top } = ref.current.getBoundingClientRect();
    setClickPosition({
      x: Math.abs(left - event.clientX),
      y: Math.abs(top - event.clientY)
    });
  };

  return (
    <div ref={ref} className={`stage ${isStageSelected() ? 'selected' : ''}`} { ...buttonize(onClick) }>
      <div className="stageCard">
        <IKImage
          src={URLS.stage(id, dlc)}
          alt={`stage: ${name}`}
          loading="lazy"
          lqip={{ active: true }}
        />
        { displayName && renderStageName() }
        { isStageSelected() && <div className="selectedStageToken">
          <IKImage
            path={'/assets/stage_coin.png'}
            alt="Stage select token"
            loading="lazy"
            lqip={{ active: true }}
            style={{
              position: 'relative',
              top: `${clickPostion.y}px`,
              left: `${clickPostion.x}px`,
            }}
          />
        </div> }
      </div>
    </div>
  );
};

const {
  bool,
  number,
  oneOfType,
  string
} = propTypes;

Stage.propTypes = {
  dlc: oneOfType([
    bool,
    number
  ]).isRequired,
  displayName: bool,
  id: number.isRequired,
  name: string.isRequired
};

Stage.defaultProps = {
  displayName: false
};

export default Stage;
