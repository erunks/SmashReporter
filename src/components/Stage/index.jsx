import React from 'react';
import propTypes from 'prop-types';
import { URLS } from '../../lib/imageAssets';

import './Stage.scss';

const Stage = (props) => (
    <div className="stage">
      <div className="stageCard">
        <img src={URLS.stage(props.id, props.dlc)} alt={`stage: ${props.name}`} />
        <div className="namePlate">
          <div className="name">{props.name}</div>
        </div>
      </div>
    </div>
);

const {
  bool,
  number,
  oneOfType,
  string
} = propTypes;

Stage.propTypes = {
  dlc: oneOfType([
    bool,
    number
  ]).isRequired,
  id: number.isRequired,
  name: string.isRequired
};

export default Stage;
