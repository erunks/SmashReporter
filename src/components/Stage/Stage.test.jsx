import React from 'react';
import { shallow } from 'enzyme';
import Stage from './index.jsx';

describe('<Stage>', () => {
  const props = {
    dlc: 0,
    id: 1,
    name: 'Random'
  };
  const container = shallow(<Stage {...props}/>);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
