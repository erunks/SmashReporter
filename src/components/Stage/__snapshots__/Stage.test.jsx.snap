// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage> should match the snapshot 1`] = `"<div class=\\"stage\\"><div class=\\"stageCard\\"><img src=\\"https://www.smashbros.com/assets_v2/img/stage/stage_img1.jpg\\" alt=\\"stage: Random\\"/><div class=\\"namePlate\\"><div class=\\"name\\">Random</div></div></div></div>"`;
